File: 0202-happy-number/0202-happy-number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        l=[]
        s=n
        while(n!=1 and n not in l):
            l.append(n)
            z=0
            while(n):
                num=n%10
                z+=(num**2)
                n//=10
            if(z==1):
                return True
            n=z
        if (n==1):
            return True
        return False  


        
        
        

File: 0202-happy-number/0202-happy-number.cpp
class Solution {
public:
    bool isHappy(int n) {
        vector<int> sn;
        while (n != 1 && find(sn.begin(), sn.end(), n) == sn.end()) {
            sn.push_back(n);
            int z = 0;
            while (n) {
                int num = n % 10;
                z += num * num;
                n /= 10;
            }
            if (z == 1) {
                return true;
            }
            n = z;
        }
        return n == 1;
    }
};

